homeassistant/components/heos/config_flow.py

```diff
@@ -185,7 +185,7 @@ async def async_step_reauth_confirm(
         errors: dict[str, str] = {}
         entry = self._get_reauth_entry()
         if user_input is not None:
-            heos = cast(Heos, entry.runtime_data.controller_manager.controller)
+            heos = cast(Heos, entry.runtime_data.coordinator.heos)
```

Comment:
I am not sure why we still need this cast

Reply:
Entry is `ConfigEntry[Any]` as returned `self._get_reauth_entry()` from  Should the cast be on entry to `HeosConfigEntry` instead?

Reply:
But when the integration is starting the reauth flow, the normal operation aborts, so then the config entry would not be loaded anymore right?

Reply:
Since auth is optional, the integration loads even when auth fails. Reauth is only issued while it's currently loaded.

Reply:
I changed the cast to be on the entry itself as that's better anyway.

Reply:
Here are the tests that specifically target this condition: https://github.com/home-assistant/core/blob/ac7b9d76395d2f0da1a46c3532cc4a47a97c86a3/tests/components/heos/test_config_flow.py#L339-L427

Reply:
I think it would be nice to add a `assert config_entry.state is ConfigEntryState.LOADED` just to make sure this is the case

Reply:
Added to the tests and in the config_flow!